src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -1,78 +1,245 @@
+        if (!waypointModel.hasAltitudeRestrictions) {
+            return;
+        }
+
+        altitudeOffsets.push([
+            waypointOffsetMap[waypointIndex],
+            Math.min(waypointModel.altitudeMaximum, waypointModel.altitudeMinimum)
+        ]);
+    });
+
+    return altitudeOffsets;
+}
+
+/**
+ * Calculate the ideal spawn altitude at the given distance along the route from the spawn point
+ *
+ * @function _calculateAltitudeAtOffset
+ * @param waypointModelList {array<WaypointModel>} array of all waypoints along the route
+ * @param waypointOffsetMap {array<number>} offset distance from spawn point to each waypoint
+ * @param altitudeOffsets {array<array<number>>} information about location of altitude restrictions
+ * @param offsetDistance {number} distance along route at which we want to know the ideal spawn altitude
+ * @return {number} ideal spawn altitude, in feet (rounded up to nearest thousand)
+ */
+function _calculateAltitudeAtOffset(waypointModelList, waypointOffsetMap, altitudeOffsets, offsetDistance) {
+    const indexOfDistance = 0;
+    const indexOfAltitude = 1;
+    const indexOfNextAltitudeRestriction = _findIndex(altitudeOffsets, (altitudeOffset) => {
+        return altitudeOffset[indexOfDistance] >= offsetDistance;
+    });
+    const indexOfPreviousAltitudeRestriction = indexOfNextAltitudeRestriction - 1;
+
+    // handle cases where we dont have altitude restrictions ahead AND behind the spawn point
+    if (indexOfNextAltitudeRestriction < 0) {
+        if (indexOfPreviousAltitudeRestriction < 0) {
+            return undefined;
+        }
+
+        return altitudeOffsets[indexOfPreviousAltitudeRestriction][indexOfAltitude];
+    } else if (indexOfPreviousAltitudeRestriction < 0) {
+        return altitudeOffsets[indexOfNextAltitudeRestriction][indexOfAltitude];
+    }
+
+    const previousAltitudeRestriction = altitudeOffsets[indexOfPreviousAltitudeRestriction];
+    const nextAltitudeRestriction = altitudeOffsets[indexOfNextAltitudeRestriction];
+    const distanceBetweenRestrictions = nextAltitudeRestriction[indexOfDistance] - previousAltitudeRestriction[indexOfDistance];
+
+    if (distanceBetweenRestrictions === 0) {
+        return previousAltitudeRestriction[indexOfAltitude];
+    }
+
+    const altitudeBetweenRestrictions = nextAltitudeRestriction[indexOfAltitude] - previousAltitudeRestriction[indexOfAltitude];
+    const distanceFromPreviousRestrictionToOffsetDistance = offsetDistance - previousAltitudeRestriction[indexOfDistance];
+    const progressBetweenRestrictions = distanceFromPreviousRestrictionToOffsetDistance / distanceBetweenRestrictions;
+    const altitudeChangeFromPreviousRestriction = altitudeBetweenRestrictions * progressBetweenRestrictions;
+
+    return previousAltitudeRestriction[indexOfAltitude] + altitudeChangeFromPreviousRestriction;
+}
+
+// function _calculateSpawnPositionsAndAltitudes(waypointModelList, spawnOffsets) {
```

Comment:
If we don’t need it, removed it

Reply:
this was to be used, just wasn't implemented yet. has since been uncommented.